compensated-ruby/lib/compensated/cli.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'compensated'
+
+module Compensated
+  # Contains the code related to running Compensated from the command line.
+  # Note: this module is _not_ included by default when you `require 'compensated'`
+  # However, it can be included if necessary via `require 'compensated/cli'
+  module CLI
+    def self.run(argv)
+      options = OptionsParser.new.options(argv)
+      Runner.new(options: options, command: ARGV[0]).run
+    end
+
+    # Runs the CLI by pulling together the configuration from the compensated.json
+    # file as well as ensuring the payment processors are loaded into the execution
+    # context.
+    class Runner
+      attr_accessor :options, :command
+      def initialize(options:, command:)
+        self.options = options
+        self.command = prepare_command(command)
+      end
+
+      def run
+        require_payment_processors
+        command.execute
+      end
+
+      private def require_payment_processors
+        config[:payment_processors].each do |payment_processor|
+          require "compensated/#{payment_processor}"
+        end
+      end
+
+      private def config
+        @config ||= Compensated.json_adapter.parse(File.read(File.absolute_path('compensated.json')))
```

Comment:
I assume you have a `compensated.json` locally. Assuming this is to allow other users to use this cli, would it be useful to have an `compensated.example.json`? 
Or we should just point users to look at `features/cross-platform-selling.feature`?

Reply:
My hope is that we will have a `compensated init` that walks through generation of a compensated.json; but for now I think pointing people at feature files that define the expected behavior is ideal.

Reply:
At least until we start figuring out a more coherent dev-rel strategy